dts/bindings/stepper/gpio-stepper-controller.yaml

```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO Stepper controller for darlington transistor arrays or dual H-bridge
+
+compatible: "gpio-stepper"
```

Comment:
cool, so yeah my last concern is some ambiguity from this to a future step/direction driver, should this have a different name? saves changing it later

Reply:
I would actually call that one step_dir_driver. The famous ones are drv8825 and a4988 and they are actually famous like that.

Reply:
I totally agree with that, they are greatly known for the step-dir interface instead of the actual hardware mechanics

Reply:
> I would actually call that one step_dir_driver. The famous ones are drv8825 and a4988 and they are actually famous like that.

That's not flexible enough (see: https://github.com/zephyrproject-rtos/zephyr/pull/68774#discussion_r1720911226). You'll talk about a "step-dir-driver" but forget, that you are able to control the step signal either by a timer/PWM (SoC-HW/Zephyr-SW) or GPIOs. On the other side, the committed "gpio-stepper" in this PR is potentially able to support 4- and 2-pin chips (vendor independent !!!). So, the view of the technology should definitely be reversed here.

- "gpio-steppers" -> for bit banging, full-/half-steps, 4-pin (A+/B+/A-/B-) and 2-pin (DIR/STP)
- "pwm-steppers" -> for timer driven, full- until 1/2^8-steps, also 4-pin (A+/B+/A-/B-) and 2-pin (DIR/STP)
- and then the specialists, e.g. "ad,tmc5041"

Incidentally, the TMC5041 is a good example of why the API must conceptually offer the management of multiple motors for all low-level chip drivers at the lower end of the API right from the start. And even this "API/Driver Mechanic" can be easy designed with simple GPIO bitbang driver.